apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,300 @@
+
+	tm_sample_handle2 = task_manager_register_task("task2", 100, 1024, itc_task_manager_pause_task1_main, NULL, TM_APP_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle2, 0);
+
+	int ret;
+
+	ret = task_manager_start(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_start", ret, OK);
+
+	ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_start", ret, OK);
+
+	sleep(1);
+
+	ret = task_manager_unregister(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+void itc_task_manager_pause_task2(void)
+{
+	int ret;
+
+	ret = task_manager_pause(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_pause", ret, OK);
+
+	ret = task_manager_resume(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_resume", ret, OK);
+
+	ret = task_manager_restart(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_restart", ret, OK);
+
+	ret = task_manager_stop(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_stop", ret, OK);
+
+	ret = task_manager_unregister(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+}
+
+int itc_task_manager_pause_task2_main(int argc, char *argv[])
+{
+	itc_task_manager_pause_task2();
+	return 0;
+}
+
+void itc_task_manager_task4(void)
+{
+	tm_sample_handle = task_manager_register_task("task5", 100, 1024, not_builtin_task_1, NULL, TM_APP_PERMISSION_ALL, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle, 0);
+
+	int ret;
+
+	ret = task_manager_start(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_register", ret, OK);
+
+	while (1) {
+		usleep(1);
+	}
+}
+
+int itc_task_manager_task4_main(int argc, char *argv[])
+{
+	itc_task_manager_task4();
+	return 0;
+}
+
+void itc_task_manager_check_all_permission_p(void)
+{
+	int ret;
+	int tm_sample_handle3;
+
+	tm_sample_handle3 = task_manager_register_task("task4", 100, 1024, itc_task_manager_task4_main, NULL, TM_APP_PERMISSION_GROUP, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle3, 0);
+
+	tm_sample_handle2 = task_manager_register_task("task3", 100, 1024, itc_task_manager_pause_task2_main, NULL, TM_APP_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register", tm_sample_handle2, 0);
+
+	ret = task_manager_start(tm_sample_handle3, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_start", ret, OK);
+
+	ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_start", ret, OK);
+
+	sleep(1);
+
+	sample_list_info = ((tm_appinfo_list_t *)task_manager_getinfo_with_name("task4", TM_RESPONSE_WAIT_INF));
+	TC_ASSERT_NEQ("task_manager_getinfo_with_name", sample_list_info, NULL);
+
+	tm_sample_handle3 = sample_list_info->task.handle;
```

Comment:
tm_sample_handle3 is already assigned from "task_manager_register_task". Why did you re-assign with even unnecessary codes?